sklearn/_build_utils/__init__.py

```diff
@@ -75,3 +75,26 @@ def maybe_cythonize_extensions(top_path, config):
             nthreads=n_jobs,
             compile_time_env={'SKLEARN_OPENMP_SUPPORTED': with_openmp},
             compiler_directives={'language_level': 3})
+
+
+def gen_from_templates(templates):
+    """Generate cython files from templates"""
+    if not isinstance(templates, list):
+        templates = [templates]
+
+    for template in templates:
+        pyxfile = template.replace('.tp', '')
+
+        if not (os.path.exists(pyxfile) and
+                os.stat(template).st_mtime < os.stat(pyxfile).st_mtime):
```

Comment:
Don't we want to generate the `.pyx` files all the time?

This suggest that we allow to manually edit the `pyx` files which seems wrong

Reply:
I did not follow the discussion when this was made. @massich do you remember why you did it that way ?

Reply:
Ok so I found a comment that have been removed at some point. It was "# if .pyx.tp is not updated, no need to output .pyx". So the goal of this condition is just to avoid unnecessary things.

It happens that it covers the sdist situation by chance.

I added the comment again because it's useful, and added a clear branch for the sdist case.
I moved the import as high as possible.